src/Detector/MkvDetector.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Selective\VideoType\Detector;
+
+use Selective\VideoType\VideoFormat;
+use Selective\VideoType\VideoType;
+use Selective\VideoType\VideoMimeType;
+use SplFileObject;
+
+/**
+ * Detector.
+ */
+final class MkvDetector implements VideoDetectorInterface
+{
+    /**
+     * MKV.
+     *
+     * @param SplFileObject $file The video file
+     *
+     * @return VideoType|null The video type
+     */
+    public function detect(SplFileObject $file): ?VideoType
+    {
+        $bytes = (string)$file->fread(4);
```

Comment:
Firstly, reading first `4` bytes to check that these bytes are `1a` `45` `df` `a3`.

Then reading next `46` bytes to check whether these bytes contain `matroska` bytes.